### ISSUE TITLE
Add "make download" target.

### ISSUE THREAD

**Issue by fwalch** [neovim/neovim#2933]
TL;DR: It would be nice to have something like `make download` that downloads and extracts all the third-party tarballs.

Rationale: Ubuntu's PPA currently doesn't package the third-party dependencies, but just compiles them in statically. On Launchpad, retrieving the third-party sources can't happen at build time, as this happens in a sandbox. Instead, the sources must be stored in a repository (in our case in `neovim/deps` under the `src` folder, which is then imported into Launchpad). To update these sources, we currently use a workaround (basically run `make deps`, then go to each dependency folder and clean the build results), executed periodically by `bot-ci`. See also neovim/bot-ci#57. This could be replaced by a simple `make download` (or CMake call/..). See also https://github.com/neovim/neovim/commit/f0ab7c2ec1518c1e94303747db098c4fec0bb56c#commitcomment-11940310.

**Comment by ZyX-I** [comment]
> Rationale: Ubuntu's PPA currently doesn't package the third-party dependencies,

I am wondering why you are using NeoVim own mechanizm for dealing with dependencies *at all*? I think that proper way would be package dependencies separately and use cmake directly without top-level makefile. It is what Gentoo ebuild is doing.

**Comment by fwalch** [comment]
That would be the proper way, yes. It's how it was done for Arch Linux as well. Unfortunately, packaging for Ubuntu is a lot more complicated, so the PPA went with the easiest solution. And even if you package the third-party dependencies, you need to get the sources from somewhere (i.e. store sources in a Launchpad branch). Not sure how that works if you're not building daily "latest `master`" packages (for lua-messagepack, I manually imported the release tarball into a bzr branch). But maybe that would be better invested time.

**Comment by justinmk** [comment]
PR welcome, otherwise I don't see this happening.
